tests/e2e/kcs/test_monitor_recovery.py

```diff
@@ -0,0 +1,945 @@
+                label=constants.MON_APP_LABEL,
+                namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            )
+            + get_deployments_having_label(
+                label=constants.MGR_APP_LABEL,
+                namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            )
+        )
+        logger.info(to_revert_patches)
+        to_revert_patches_path = []
+        for dep in to_revert_patches:
+            to_revert_patches_path.append(
+                join(self.backup_dir, dep["metadata"]["name"] + ".yaml")
+            )
+        return to_revert_patches_path
+
+    def get_ceph_keyrings(self):
+        """
+        Gets all ceph and csi related keyring from OCS secrets
+
+        """
+        mon_k = get_ceph_caps(["rook-ceph-mons-keyring"])
+        if config.ENV_DATA["platform"] == constants.ON_PREM_PLATFORMS:
+            rgw_k = get_ceph_caps(
+                ["rook-ceph-rgw-ocs-storagecluster-cephobjectstore-a-keyring"]
+            )
+        else:
+            rgw_k = None
+        mgr_k = get_ceph_caps(["rook-ceph-mgr-a-keyring"])
+        mds_k = get_ceph_caps(
+            [
+                "rook-ceph-mds-ocs-storagecluster-cephfilesystem-a-keyring",
+                "rook-ceph-mds-ocs-storagecluster-cephfilesystem-b-keyring",
+            ]
+        )
+        crash_k = get_ceph_caps(["rook-ceph-crash-collector-keyring"])
+        fs_node_k = get_ceph_caps([constants.CEPHFS_NODE_SECRET])
+        rbd_node_k = get_ceph_caps([constants.RBD_NODE_SECRET])
+        fs_provisinor_k = get_ceph_caps([constants.CEPHFS_PROVISIONER_SECRET])
+        rbd_provisinor_k = get_ceph_caps([constants.RBD_PROVISIONER_SECRET])
+
+        keyring_caps = {
+            "mons": mon_k,
+            "rgws": rgw_k,
+            "mgrs": mgr_k,
+            "mdss": mds_k,
+            "crash": crash_k,
+            "fs_node": fs_node_k,
+            "rbd_node": rbd_node_k,
+            "fs_provisinor": fs_provisinor_k,
+            "rbd_provisinor": rbd_provisinor_k,
+        }
+
+        for secret, caps in keyring_caps.items():
+            if caps:
+                with open(f"{self.keyring_dir}/{secret}.keyring", "w") as fd:
+                    fd.write(caps)
+                    self.keyring_files.append(f"{self.keyring_dir}/{secret}.keyring")
+
+    def mds_deployments_to_revert(self):
+        """
+        Gets only mds deployments to revert it
+
+        Returns:
+            list: list of deployment paths to be reverted
+
+        """
+        to_revert_mds = get_deployments_having_label(
+            label=constants.MDS_APP_LABEL,
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+        )
+        to_revert_mds_path = []
+        for dep in to_revert_mds:
+            logger.info(dep)
+            to_revert_mds_path.append(
+                join(self.backup_dir, dep["metadata"]["name"] + ".yaml")
+            )
+        return to_revert_mds_path
+
+    def patch_sleep_on_mds(self):
```

Comment:
This entire code looks similar to patch_sleep_on_osds() and patch_sleep_on_mon()
Should we create a single functions which handles all?

Reply:
There are many minor changes in each patch like mon will be only patched for sleep and livenessprobe is not removed unlike other pods. It is better this way for readability and making any change in individual step at least for now.